keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/PlaceHolderDoubleVertex.java

```diff
@@ -0,0 +1,19 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.NonProbabilistic;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+
+public class PlaceHolderDoubleVertex extends DoubleVertex implements NonProbabilistic<DoubleTensor> {
```

Comment:
Interesting idea. Two questions. 1) Could we build the LogProbGraph using the hyperparameters rather than placeholders? This might offer some advantages for autodiffing the rate of change of logProb with respect to the value of upstream latent vertices. Could even have the sum operation in the graph. 2) Is there a technical reason why ConstantDoubleVertex couldn't be used as a placeholder? Appreciate that it wouldn't be as nice to instantiate with a value that would always be overwritten.

Reply:
+1 to "Is there a technical reason why ConstantDoubleVertex couldn't be used as a placeholder"

Reply:
No technical reason at the moment but I'd say a ConstantVertex shouldn't be changed by an algorithm where as a PlaceHolder requires the algorithm to give it a value. Should we disallow a ConstantVertex value change after construction?

Reply:
Do we even require our Constant Vertices to be constant (i.e. immutable)? Suppose we renamed them to something like `SimpleVertex`, `ValueVertex` or `VariableVertex` - would you then be happy to use them here too?

Reply:
A variable and constant are semantically different so I think you need to be able to express both of them separately. We could call these place holder vertices variable vertices but those two concepts will also need to be separate when we start adding in control flow where place holders would be function arguments and variables would be stateful.